src/VisualStudio/Core/Impl/Options/GridOptionPreviewControl.xaml.cs

```diff
@@ -93,5 +106,264 @@ internal override void Close()
                 this.ViewModel.Dispose();
             }
         }
+
+        internal void Generate_Save_Editorconfig(object sender, System.Windows.RoutedEventArgs e)
+        {
+            var optionSet = this.ViewModel.ApplyChangedOptions(this.OptionService.GetOptions());
+            var editorconfig = new StringBuilder();
+            Generate_Editorconfig(optionSet, _language, editorconfig, _getLangaugeSpecificEditorConfigOptions);
+            var sfd = new System.Windows.Forms.SaveFileDialog
+            {
+                Filter = "All files (*.*)|",
+                FileName = ".editorconfig",
+                Title = "Save .editorconfig file"
+            };
+            if (sfd.ShowDialog() == System.Windows.Forms.DialogResult.OK)
+            {
+                IOUtilities.PerformIO(() =>
+                {
+                    var path = sfd.FileName;
+                    using (var sw = new StreamWriter(File.Create(path)))
+                    {
+                        sw.Write(editorconfig);
+                        sw.Close();
```

Comment:
Close should be unnecessary

Reply:
File.Create should probably be in another using above this one

Reply:
nit: I don't like all these abbreviations, especially `sfd` but I'm not sure how others feel about that

Reply:
There's also just: File.WriteAllText.